src/app/shared/code-mirror/code-mirror.component.ts

```diff
@@ -32,11 +33,30 @@ export class Class implements AfterContentInit, OnChanges, OnDestroy {
 	@Input() codeMirror:CodeMirror.Editor;
 	@Output() codeMirrorChange:EventEmitter<CodeMirror.Editor> = new EventEmitter<CodeMirror.Editor>();
 
+	//TODO: this is a shortcut that could be enable to prettify, at this moment we decide disabled.
+	/*@HostListener( 'keydown', [ '$event' ] ) onKeyDown( e ) {
+		if( e.shiftKey && e.ctrlKey && e.keyCode == 80 ) {
+			try {
+				let parsedQuery = this.parser.parse( this.value );
+				this.codeMirror.setValue( this.generator.stringify( parsedQuery ) );
+			} catch( e ) {
+			}
+		}
+	}*/
+
 	private internallyChanged:boolean = false;
 	private lastUpdates:string[] = [];
+	private parser;
+	private generator;
+	private textMarkers = [];
+	private renderer:Renderer2;
 
-	constructor( element:ElementRef ) {
+	constructor( element:ElementRef, renderer:Renderer2 ) {
 		this.element = element;
+		this.parser = new SPARQL.Parser();
```

Comment:
SPARQL specific functionality shouldn't be baked into a generic component like `code-mirror.component`. This functionality is only relevant to the SPARQL client and thus should be added somewhere over there.